docs/SpellExample.scala

```diff
@@ -0,0 +1,277 @@
+    ))
+    val key: String = keyResults("secret").to[String].get
+    val client = adminClient.sessionClient(key)
+
+    try
+    {
+      println(s"Connected to Fauna database $DB_NAME with server role\n")
+      runSpellExamples(DB_NAME, client)
+    }
+    finally  {
+      client.close()
+    }
+
+    /*
+     * Just to keep things neat and tidy, delete the database and close the client connection
+     */
+    deleteDB(adminClient, DB_NAME)
+    adminClient.close()
+    println("Disconnected from FaunaDB as Admin!")
+  }
+
+  private def runSpellExamples(DB_NAME: String, client: FaunaClient): Unit = {
+
+    /*
+    * Create the spell class and index
+    */
+    val SPELLS_CLASS = "spells"
+    val INDEX_NAME = "spells_index"
+
+    val classResults: Value = await(client.query(CreateClass(Obj("name" -> SPELLS_CLASS))))
+    println(s"Create Class for $DB_NAME:\n $classResults\n")
+
+    val indexResults: Value = await(client.query(
+      CreateIndex(
+        Obj("name" -> INDEX_NAME,
+          "source" -> Class(SPELLS_CLASS)
+        )
+      )
+    ))
+    println(s"Create Index for $DB_NAME:\n $indexResults\n")
+
+    /*
+    * Add some entries to the spells class
+    */
+    val addFireResults = await(client.query(
+      Create(Class(Value(SPELLS_CLASS)),
+        Obj("data" ->
+          Obj("name" -> "Fire Beak", "element" -> "water", "cost" -> 15)))
+    ))
+
+    println(s"Added spell to class $SPELLS_CLASS: \n $addFireResults \n")
+
+    val addDragonResults = await(client.query(
+      Create(Class(Value(SPELLS_CLASS)),
+        Obj("data" ->
+          Obj("name" -> "Water Dragon's Claw", "element" -> "water")
+        )
+      )
+    ))
+    println(s"Added spell to class $SPELLS_CLASS \n $addDragonResults \n")
+
+    val addHippoResults = await(client.query(
+      Create(Class(Value(SPELLS_CLASS)),
+        Obj("data" ->
+          Obj("name" -> "Hippo's Wallow", "element" -> "water", "cost" -> 35)))
+    ))
+    println(s"Added spell to class $SPELLS_CLASS:\n $addHippoResults \n")
+
+    //The results at 'ref' are a resource pointer to the class that was just created.
+    val hippoRef = addHippoResults("ref")
+    println(s"hippoRef = $hippoRef \n")
+
+    /*
+    * Read the hippo back that we just created
+    */
+    val getHippoResults: Value = await(client.query(
+      Select(Value("data"), Get(hippoRef))
+    ))
+    println(s"Hippo Spell:\n $getHippoResults \n")
+
+    //convert the hippo results into primitive elements
+    val name: String = getHippoResults("name").to[String].get
+    val cost: Int = getHippoResults("cost").to[Int].get
+    val element: String = getHippoResults("element").to[String].get
+    println(s"Spell Details: Name=$name, Cost=$cost, Element=$element")
+
+    //This would return an empty option if the field is not found or if the conversion fails
+    val optSpellElement: Option[String] = getHippoResults("element").to[String].toOpt
```

Comment:
ok, if the `element` was absent in the result, it would have thrown an exception on the line 143

Reply:
How would you recommend doing this?   Or is there a safe way of doing this without using a try / catch?